lib/jasmine/jasmine.ts

```diff
@@ -1,32 +1,97 @@
+  const jasmineEnv = jasmine.getEnv();
+  ['desribe', 'xdescribe', 'fdescribe'].forEach((methodName) => {
+    var delegate: Function = jasmineEnv[methodName];
+    jasmineEnv[methodName] = function(description: string, specDefinitions: Function) {
+      return delegate.call(this, description,  wrapDescribeInZone(specDefinitions));
+    } 
+  });
+  ['it', 'xit', 'fit'].forEach((methodName) => {
+    var delegate: Function = jasmineEnv[methodName];
+    jasmineEnv[methodName] = function(description: string, specDefinitions: Function) {
+      return delegate.call(this, description, wrapTestInZone(specDefinitions));
+    } 
+  });
+  ['beforeEach', 'afterEach'].forEach((methodName) => {
+    var delegate: Function = jasmineEnv[methodName];
+    jasmineEnv[methodName] = function(specDefinitions: Function) {
+      return delegate.call(this, wrapTestInZone(specDefinitions));
+    } 
+  });
+
+  function wrapDescribeInZone(fn: Function): Function {
+    // Used for executing the describe content. No async work is allowed here.
+    return function() {
+      return syncZone.run(fn, this, arguments as any as any[]);
+    }
   }
 
-  function QueueRunner(options) {
-    options.clearStack = alwaysClearStack;
-    SuperQueueRunner.call(this, options);
+  function wrapTestInZone(fn: Function): Function {
+    // Used for executing it/beforeEach/afterEach. This will run in `testProxyZone`.
+    // The `testProxyZone` will be reset by the `ZoneQueueRunner`
+    if(fn.length == 0) {
+      return function() {
+        return testProxyZone.run(fn, this);
+      }
+    } else {
+      return function(done) {
+        return testProxyZone.run(fn, this, [done]);
+      }
+    }
   }
-  QueueRunner.prototype = SuperQueueRunner.prototype;
-  return QueueRunner;
-})((<any>jasmine).QueueRunner);
+  interface QueueRunner {
+    execute(): void;
+  }
+  interface QueueRunnerAttrs {
+    queueableFns: {fn: Function}[];
+    onComplete: () => void;
+    clearStack: (fn) => void;
+    onException: (error) => void;
+    catchException: () => boolean;
+    userContext: any;
+    timeout: {setTimeout: Function, clearTimeout: Function};
+    fail: ()=> void;
+  }
+
+  const QueueRunner = (jasmine as any).QueueRunner as { new(attrs: QueueRunnerAttrs): QueueRunner };
+  (jasmine as any).QueueRunner = class ZoneQueueRunner extends QueueRunner {
+    constructor(attrs: QueueRunnerAttrs) {
+      attrs.clearStack = (fn) => fn(); // Don't clear since onComplete will clear.
+      attrs.onComplete = ((fn) => () => {
+        // All functions are done, clear the test zone.
+        testProxyZone = null;
+        ambientZone.scheduleMicroTask('jasmine.onComplete', fn);
+      })(attrs.onComplete);
+      super(attrs);
+    }
 
+    execute() {
+      if(Zone.current !== ambientZone) throw new Error("Unexpected Zone: " + Zone.current.name);
+      testProxyZone = ambientZone.fork(new ProxyZoneSpec());
```

Comment:
Good catch, but no. The execute calls the functions which have been patched above. So there is no need for calling it in the test zone.